v0/destinations/campaign_manager/transform.js

```diff
@@ -97,13 +168,32 @@ function validateRequest(message) {
       400
     );
   }
+}
 
+function postValidateRequest(response) {
+  // response.body.JSON.conversion[0]
   if (
-    message.properties.encryptedUserId &&
-    !message.properties.encryptionInfo
+    (response.body.JSON.conversions[0].encryptedUserId ||
+      response.body.JSON.conversions[0].encryptedUserIdCandidates) &&
+    !response.body.JSON.encryptionInfo
   ) {
     throw new CustomError(
-      "[CAMPAIGN MANAGER (DCM)]: encryptionInfo is a required field if encryptedUserId is used.",
+      "[CAMPAIGN MANAGER (DCM)]: encryptionInfo is a required field if encryptedUserId or encryptedUserIdCandidates is used.",
+      400
+    );
+  }
+
+  if (
+    !response.body.JSON.conversions[0].gclid &&
+    !response.body.JSON.conversions[0].dclid &&
+    !response.body.JSON.conversions[0].encryptedUserId &&
+    !response.body.JSON.conversions[0].encryptedUserIdCandidates &&
+    !response.body.JSON.conversions[0].matchId &&
+    !response.body.JSON.conversions[0].mobileDeviceId &&
+    !response.body.JSON.conversions[0].impressionId
```

Comment:
better to write it like this
```suggestion
    !(response.body.JSON.conversions[0].gclid ||
    response.body.JSON.conversions[0].dclid ||
    response.body.JSON.conversions[0].encryptedUserId ||
    response.body.JSON.conversions[0].encryptedUserIdCandidates ||
    response.body.JSON.conversions[0].matchId ||
    response.body.JSON.conversions[0].mobileDeviceId ||
    response.body.JSON.conversions[0].impressionId)
```